lib/factory_bot/definition.rb

```diff
@@ -191,5 +193,14 @@ def automatically_register_defined_enums?(klass)
       FactoryBot.automatically_define_enum_traits &&
         klass.respond_to?(:defined_enums)
     end
+
+    def ensure_attribute_not_generating_primary_key!(attribute, for_class:)
+      case attribute
+      when FactoryBot::Attribute::Sequence
```

Comment:
While I'm skeptical of a legitimate (intentional!) use case for taking over control of a table's primary key, I think it might be helpful to provide an override for test suites that want to.

In the circumstance, would it be helpful to **encourage the use** of an escape hatch that combines a dynamic attribute and a call to `generate`?

```ruby
sequence(:a_primary_key_column_name)

# ...

id { generate(: a_primary_key_column_name) }
```